### PR TITLE
update xero-accounts-last-synced-at

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/workspaces/utils.py [hunk 1]
```diff
@@ -127,6 +129,10 @@ def create_or_update_general_settings(general_settings_payload: Dict, workspace_
         }
     )
 
+    if set(workspace_general_settings.charts_of_accounts) != set(general_settings_payload['charts_of_accounts']):
+        workspace.xero_accounts_last_synced_at = datetime.datetime.now() - datetime.timedelta(days=5*365)
```
Comment (ashwin1111) [inline apps/workspaces/utils.py:133]: let's set it to null, let's not assume any past dates

File: apps/xero/utils.py [hunk 2]
```diff
@@ -203,6 +214,9 @@ def sync_accounts(self):
                 DestinationAttribute.bulk_create_or_update_destination_attributes(
                     account_attribute, attribute_type.upper(), self.workspace_id, True)
 
+        workspace.xero_accounts_last_synced_at = get_last_synced_at(self.workspace_id, 'ACCOUNT')
```
Comment (ashwin1111) [inline apps/xero/utils.py:217]: why not set to current time directly?

File: apps/xero/utils.py [hunk 3]
```diff
@@ -163,9 +166,17 @@ def sync_accounts(self):
 
         self.connection.set_tenant_id(tenant_mapping.tenant_id)
 
-        updated_at = self.__get_last_synced_at('ACCOUNT')
+        workspace = Workspace.objects.filter(id=self.workspace_id).first()
+
+        if workspace.xero_accounts_last_synced_at:
+            updated_at = format_updated_at(workspace.xero_accounts_last_synced_at)
+        else:
```
Comment (ashwin1111) [inline apps/xero/utils.py:173]: since we are setting xero_accounts_last_synced_at to null, else case is not needed

File: apps/xero/utils.py [hunk 4]
```diff
@@ -41,18 +56,6 @@ def __init__(self, credentials_object: XeroCredentials, workspace_id: int):
         credentials_object.refresh_token = self.connection.refresh_token
         credentials_object.save()
 
-    def __format_updated_at(self, updated_at):
```
Comment (ashwin1111) [inline apps/xero/utils.py:44]: any reason why this was moved outside class?

Comment (Sravanksk) [reply apps/xero/utils.py:44]: has to do that because it is imported in workspace utils - also pycharm suggestions - let it be

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**